### PR TITLE
[BUGFIX] Tooltip locale errors can cause chart to break on hover

### PR DESCRIPTION
**Pull request by sjcobb**
This PR fixes a Firefox locale timezone issue where our graph can break completely on hover. 

Settings example:
image

## Before
image

On hover over 'Queue Length' line chart:
image

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky. See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests) and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: ui/components/src/TimeSeriesTooltip/TooltipContent.tsx [hunk 1]
```diff
@@ -26,9 +26,9 @@ export function TooltipContent(props: TooltipContentProps) {
   const { focusedSeries, wrapLabels } = props;
   const { formatWithUserTimeZone } = useTimeZone();
 
-  const seriesTime = focusedSeries && focusedSeries[0] && focusedSeries[0].date ? focusedSeries[0].date : null;
+  const seriesTime = focusedSeries && focusedSeries[0] && focusedSeries[0].dateMs ? focusedSeries[0].dateMs : null;
 
-  const formatTimeSeriesHeader = (timeString: string) => {
+  const formatTimeSeriesHeader = (timeString: number) => {
```
Comment (cndonovan) [inline ui/components/src/TimeSeriesTooltip/TooltipContent.tsx:31]: Can rename back to `time`!

File: ui/components/src/TimeSeriesTooltip/TooltipContent.test.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ describe('TooltipContent', () => {
     renderComponent(tooltipContent);
     expect(screen.getByText('Test node demo.do.prometheus.io:9100')).toBeInTheDocument();
     expect(screen.getByText('Dec 23, 2022 -')).toBeInTheDocument();
-    expect(screen.getByText('13:53:00')).toBeInTheDocument();
+    expect(screen.getByText('18:53:00')).toBeInTheDocument();
```
Comment (cndonovan) [inline ui/components/src/TimeSeriesTooltip/TooltipContent.test.tsx:42]: This was supposed to change, I assume?

File: ui/components/src/TimeSeriesTooltip/TooltipContent.test.tsx [hunk 3]
```diff
@@ -27,8 +27,8 @@ describe('TooltipContent', () => {
           seriesIdx: 0,
           datumIdx: 84,
           seriesName: 'Test node demo.do.prometheus.io:9100',
-          date: 'Dec 23, 2022, 1:53:00 PM',
-          x: 1671821580000,
+          dateMs: 1671821580000,
+          x: 1671821580,
```
Comment (cndonovan) [inline ui/components/src/TimeSeriesTooltip/TooltipContent.test.tsx:31]: Wonder why the `x`  value changed for this one, but not the other tests

**Review by cndonovan — APPROVED** [review]
This might be silly b/c you didn't ask for review yet, but I'm just up late poking at stuff so I figured I'd give thumbs up

Way to solve a Firefox bug, browser-specific bugs can be so mysterious

File: ui/components/src/TimeSeriesTooltip/TooltipContent.test.tsx (hunk 3 above)
Comment (sjcobb) [reply ui/components/src/TimeSeriesTooltip/TooltipContent.test.tsx:31]: I'm going to move this to a separate unit test to make this more explicit since this isn't directly related to this PR, sorry I should have commented about it! 

While in this code, I remembered [this edge case](https://github.com/perses/perses/blob/v0.29.1/ui/components/src/TimeSeriesTooltip/focused-series.ts#L63-L65)  in focused-series.ts. One place where we were consuming LineChart internally, didn't pass x values as milliseconds and I still wanted the tooltip to work so I added a condition that checked for this. There isn't a unit test though so wanted to confirm that functionality still worked, but I'll update this to be a new test case!

Comment (sjcobb) [reply ui/components/src/TimeSeriesTooltip/TooltipContent.test.tsx:31]: I need to check if this is even an issue internally anymore, I think the results _are_ being converted to milliseconds now, so maybe [that conversion](https://github.com/perses/perses/blob/v0.29.1/ui/components/src/TimeSeriesTooltip/focused-series.ts#L64) isn't really needed 🤔

**Review by eunicorn — APPROVED** [review]

File: ui/components/src/TimeSeriesTooltip/TooltipContent.test.tsx (hunk 2 above)
Comment (cndonovan) [reply ui/components/src/TimeSeriesTooltip/TooltipContent.test.tsx:42]: Euuuuuniceee with the 👀

**Comment by sjcobb** [comment]
Closing in favor of the merged fix in #1163
